### DISCUSSION TITLE
DNS not working on CentOS Stream 9

### DISCUSSION THREAD

**Discussion by francoism90** [containers/podman#22725] [Q&A] [answered]
```
[Unit]
Description=App container

[Service]
EnvironmentFile=%h/.config/containers/systemd/config/app.env
TimeoutStartSec=900
Restart=always

[Container]
Image=localhost/my-app
AutoUpdate=local
Volume=${APP_PATH}:/app:rw,z,U
User=${APP_UID}
Group=${APP_GID}
UserNS=keep-id:uid=${APP_UID},gid=${APP_GID}
Network=internal.network
ExposeHostPort=9000

[Install]
WantedBy=multi-user.target default.target
```
I just cannot connect to any external upstream DNS-records.

I also tried using the latest unstable branch.
Is this a `pasta` problem?

```
$ podman -v
podman version 5.1.0-dev-c4631dc2d
```

**Comment by sbrivio-rh** [comment]
Can you describe the environment a bit? What DNS server are you trying to contact, how, and what resolvers do you have configured in host and container?

**Reply by sbrivio-rh** [reply]
...also, what version of `passt` do you have installed?

**Reply by francoism90** [reply]
@sbrivio-rh Thanks for looking into this. :)

It's inside the container; it doesn't seem to be able to contact any external URLs/requests - internal requests work fine. When switching back to `slirp4netns` it works fine again.

I do not have any resolvers configured, it should use the host/system one as fallback, right?

These are the packages installed:

```bash
$ pasta --version
pasta 0^20231204.gb86afe3-1.el9.x86_64

$ podman -v
podman version 5.1.0-dev-c4631dc2d
```

**Reply by sbrivio-rh** [reply]
> I do not have any resolvers configured, it should use the host/system one as fallback, right?

Not just as fallback -- pasta would simply map requests to the one on the host. I guess you have a resolver configured on the host: I was wondering about its address, because we had some issues in the past with handling of loopback addresses (systemd-resolved).

> These are the packages installed:

I had a particular issue in mind with previous versions of passt, but not with that one. Could you anyway try, as a test, with an unofficial, newer version from my Copr repository? Links here: https://copr.fedorainfracloud.org/coprs/sbrivio/passt/build/7434838/

**Reply by francoism90** [reply]
@sbrivio-rh Thanks! This seems to fix my issue. I need to test internal resolving, but external works (like composer install/npm install).

I think CentOS should push a newer version. Should I report this?

**Reply by francoism90** [reply]
For the record, this is what I have now installed on CentOS Stream 9:
```bash
$ pasta --version
pasta 0^20240607.g8a83b53-1.el9.x86_64

$ podman -v
podman version 5.1.0
```
I have reinstalled the VPS, so with the default pasta version, it wouldn't work. Using your latest provided build (I didn't tested the older you linked at first), it works as it should. :)

**Reply by sbrivio-rh** [reply, marked as answer]
Thanks @francoism90 for testing!

> I think CentOS should push a newer version. Should I report this?

No, actually, I already [rebased](https://gitlab.com/redhat/centos-stream/rpms/passt/-/merge_requests/15) the CentOS Stream 9 package to a new version a while ago, but due to a mistake on my side the updated package didn't hit the repositories. It will be available in a bit.

**Reply by francoism90** [reply]
@sbrivio-rh Thanks for the help! I did already report it upstream, can you manage the ticket?

I can also close it, but I don't know if you need to reference this. :)

**Reply by sbrivio-rh** [reply]
> @sbrivio-rh Thanks for the help! I did already report it upstream, can you manage the ticket?

I didn't notice any ticket yet... where did you report it?

**Reply by sbrivio-rh** [reply]
Oh, I found it now, https://issues.redhat.com/browse/RHEL-44341.

**Reply by francoism90** [reply]
@sbrivio-rh Sorry for asking, but I'm still seeing the old packages:
```bash
Last metadata expiration check: 0:00:10 ago on Thu 27 Jun 2024 09:56:02 AM UTC.
Dependencies resolved.
===================================================================================================================================================================================
 Package                                   Architecture                       Version                                                  Repository                             Size
===================================================================================================================================================================================
Installing:
 passt                                     x86_64                             0^20231204.gb86afe3-1.el9                                appstream                             186 k
 passt-selinux                             noarch                             0^20231204.gb86afe3-1.el9                                appstream                              31 k
```
How long does it usually take before the new packages appear in the repo?

Thanks!

**Reply by sbrivio-rh** [reply]
> @sbrivio-rh Sorry for asking, but I'm still seeing the old packages:

I know, sorry for that. There definitely seems to be a problem that I'm already investigating. I'll get back to you as soon as the packages are available.

**Reply by sbrivio-rh** [reply]
[passt-0^20240624.g1ee2eca-1.el9](https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/passt-0%5e20240624.g1ee2eca-1.el9.x86_64.rpm) is now available in the mirrors.

**Reply by francoism90** [reply]
@sbrivio-rh Thanks!
